src/algorithms/calorimetry/ImagingTopoCluster.cc

```diff
@@ -50,12 +50,28 @@ void ImagingTopoCluster::init() {
 
   // unitless conversion
   // sanity checks
-  if (m_cfg.localDistXY.size() != 2) {
-    error("Expected 2 values (x_dist, y_dist) for localDistXY");
+  if (m_cfg.sameLayerDistXY.size() != 2) {
+    error("Expected 2 values (x_dist, y_dist) for sameLayerDistXY");
     return;
   }
-  if (m_cfg.layerDistEtaPhi.size() != 2) {
-    error("Expected 2 values (eta_dist, phi_dist) for layerDistEtaPhi");
+  if (m_cfg.diffLayerDistXY.size() != 2) {
+    error("Expected 2 values (x_dist, y_dist) for diffLayerDistXY");
+    return;
+  }
+  if (m_cfg.sameLayerDistEtaPhi.size() != 2) {
+    error("Expected 2 values (eta_dist, phi_dist) for sameLayerDistEtaPhi");
+    return;
+  }
+  if (m_cfg.diffLayerDistEtaPhi.size() != 2) {
+    error("Expected 2 values (eta_dist, phi_dist) for diffLayerDistEtaPhi");
+    return;
+  }
+  if (m_cfg.sameLayerDistPhiZ.size() != 2) {
+    error("Expected 2 values (phi_dist, z_dist) for sameLayerDistPhiZ");
+    return;
+  }
+  if (m_cfg.diffLayerDistPhiZ.size() != 2) {
+    error("Expected 2 values (phi_dist, z_dist) for diffLayerDistPhiZ");
     return;
   }
```

Comment:
We could move these into the the switch statements below to not require all values to be specified. Also consider replacing `return` with `throw std::runtime_error("<Same message as in error()>")`.